### PR TITLE
Institutions api

### PR DESCRIPTION
**Pull request by fego-dev**
# Description
What did you implemented/Why did you implemented this:
 - Add search and index methods for institutions api
 - I restrict the data to pass to the api consumers
 - I add search route for institutions api. 
 - I fix aplication controller for query to api without a sesion, this is temporaly

### CODE REVIEW

**Review by MiguelOcegueraM — APPROVED** [review]
The code is very clean, good job mates

File: db/schema.rb [hunk 1]
```diff
@@ -15,18 +15,6 @@
...
-    t.string "email", default: "", null: false
-    t.string "encrypted_password", default: "", null: false
-    t.string "reset_password_token"
-    t.datetime "reset_password_sent_at"
-    t.datetime "remember_created_at"
-    t.datetime "created_at", precision: 6, null: false
-    t.datetime "updated_at", precision: 6, null: false
-    t.index ["email"], name: "index_admins_on_email", unique: true
-    t.index ["reset_password_token"], name: "index_admins_on_reset_password_token", unique: true
-  end
```
Comment (EduardoGHdez) [inline db/schema.rb:28]: Did you manually remove this table?
Was there any migration to do this?

File: config/routes.rb [hunk 2]
```diff
@@ -18,14 +17,16 @@
...
 	  resources :user_history_details
 	  resources :user_histories
 	  resources :area_histories
 	  resources :areas
 	  resources :symptoms
-	  resources :insititutions
+	  resources :institutions
 	  devise_for :users
+
+    post 'institutions/search', :to => 'institutions#search'
```
Comment (EduardoGHdez) [inline config/routes.rb:29]: POST is supposed to be used to mutate data in the DB.
WDTY about using `GET` here?

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,5 +1,5 @@
 class ApplicationController < ActionController::Base
-
+  protect_from_forgery with: :null_session
```
Comment (EduardoGHdez) [inline app/controllers/application_controller.rb:2]: I think this is common for API's

WDYT to do this only in API controllers?

File: db/schema.rb (hunk 1 above)
Comment (fego-dev) [reply db/schema.rb:28]: Hi Eduardo, this by mistake went to the master, it should not be there, there is no such model.

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (fego-dev) [reply app/controllers/application_controller.rb:2]: I understand, but the base application controller is shared by the api and web controllers, that's why it's there.

File: config/routes.rb (hunk 2 above)
Comment (fego-dev) [reply config/routes.rb:29]: Sorry my bad, you are right, I must change it, thanks for your comments

File: db/schema.rb (hunk 1 above)
Comment (EduardoGHdez) [reply db/schema.rb:28]: If not, I'd suggest using a migration instead of manually editing the `schema`

Comment (fego-dev) [reply db/schema.rb:28]: Thank you for your advice, we will take it into account so as not to fall into the same error.

**Review by LPMendoza — APPROVED** [review]
Good job, continue like that!

File: app/views/layouts/_sidebar.html.erb [hunk 4]
```diff
@@ -41,6 +41,14 @@
               </p>
             </a>
           </li>
+          <li class="nav-item">
+            <a href="/areas" class="nav-link">
+              <i class="nav-icon fas fa-map-marker-alt"></i>
+              <p>
+                Areas
```
Comment (OchoaHub) [inline app/views/layouts/_sidebar.html.erb:48]: Ty for add areas in the adminlte;)

File: app/serializers/institution_serializer.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class InstitutionSerializer
+  include FastJsonapi::ObjectSerializer
```
Comment (OchoaHub) [inline app/serializers/institution_serializer.rb:2]: I like this gem, thanks for sharing!

**Review by OchoaHub — APPROVED** [review]

**Pull request merged**